src/init/features/frameworks/repo.ts

```diff
@@ -81,6 +83,10 @@ export async function linkGitHubRepository(
   logger.info(clc.bold(`\n${clc.yellow("===")} Connect a GitHub repository`));
   const existingConns = await listFrameworksConnections(projectId);
   if (existingConns.length < 1) {
```

Comment:
This may not be the best sentinel for whether the SA is set up properly. Users can always go into their IAM policy and remove the role grant. The more futureproof way of doing this is to read their IAM policy and verify the service account is still provisioned correctly.

I'm happy to leave it as is for M2 but we should probably note this in a line comment or track in a bug

Reply:
> This may not be the best sentinel for whether the SA is set up properly. Users can always go into their IAM policy and remove the role grant.

AFAIK, this is the only code path right now where we need the IAM grant on the GCB service account.  If we are reusing an existing connection, there is no need to make an IAM grant. It's only when setting up the connection for the very first time when we need the IAM grant.

A sub-goal here was to make sure we don't ask for granting this IAM policy unless it's necessary since it's yet another friction for onboarding.